src/shared/conf-parser.c

```diff
@@ -725,6 +726,36 @@ int config_parse_tristate(
         return 0;
 }
 
+int config_parse_io_cost_ctrl(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+        IOCostCtrl *ctrl = data;
+        IOCostCtrl io_cost_ctrl;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        io_cost_ctrl = io_cost_ctrl_from_string(rvalue);
+        if (!io_cost_ctrl) {
```

Comment:
io_cost_ctrl_from_string returns -EINVAL on error, not 0. i.e. test with `if (io_cost_ctrl < 0)` for errors